### PR TITLE
feat: add fmt::Display implementation for path data structs

### LINKED ISSUE #2181: Improve display of object store path names
__Use case:__
While logging, such as https://github.com/influxdata/influxdb_iox/pull/2180 it would be nice to have human readable display paths in the logs to make it easier to understand

__Proposal:__
Implement `std::fmt::Display` for `object_store::Path`:

__Current behaviour:__
Paths in the logs look like this
```
location=File(FilePath { inner: Parsed(DirsAndFileName { directories: [PathPart("42")], file_name: None })
```

__Desired behaviour:__
```
location="/42"
```

### PR DESCRIPTION
**Pull request by sunng87**
Closes #2181 

Added `fmt::Display` implementations for `Path` and related types. For `Path`, prefix like `s3:`/`file:` was added to distinguish its backend.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Comment by crepererum** [comment]
I'm wondering if we wanna remove the `Path::display` method then. It is used in some places but could then easily replaced w/ `Path::to_string` (which is now indirectly implemented via the new `Display` impl). We can do that in a separate PR though.

**Review by alamb — COMMENTED** [review]
Thank you @sunng87 !

File: object_store/src/path.rs [hunk 1]
```diff
@@ -128,3 +130,71 @@ impl From<Path> for DirsAndFileName {
...
+    use std::path::PathBuf;
+
+    use super::*;
+    use crate::parsed_path;
+
+    #[test]
+    fn test_display() {
+        assert_eq!(
+            format!("{}", Path::AmazonS3(CloudPath::raw("/foo/bar/baz.json"))),
+            "s3:/foo/bar/baz.json"
```
Comment (alamb) [inline object_store/src/path.rs:167]: ❤️

**Comment by sunng87** [comment]
@crepererum I will open a new pull request for that refactoring job.

**Review by alamb — APPROVED** [review]
Thanks again @sunng87  -- I agree that removing the `display()` method would be a nice cleanup as a follow on PR

**Comment by alamb** [comment]
I have validated that the CLA is signed. Thanks!

**Pull request merged**